### PR TITLE
python3Packages.pytorch-metric-learning: init at 0.9.81

### PR DESCRIPTION
**Pull request by bcdarwin**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [NA] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [NA] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [NA] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/pytorch-metric-learning/default.nix
Comment (drewrisinger) [inline pkgs/development/python-modules/pytorch-metric-learning/default.nix]: Break into separate lines & sort alphabetically.

File: pkgs/development/python-modules/pytorch-metric-learning/default.nix
Comment (drewrisinger) [inline pkgs/development/python-modules/pytorch-metric-learning/default.nix]:
```suggestion
  disabled = !isPy3k;
```
Both should work fine.

File: pkgs/development/python-modules/pytorch-metric-learning/default.nix
Comment (drewrisinger) [inline pkgs/development/python-modules/pytorch-metric-learning/default.nix]: Move lib to top by convention

**Review by drewrisinger — CHANGES_REQUESTED** [review]
Style looks pretty good, haven't tested build.

**Review by jonringer — APPROVED** [review]
LGTM (after other suggestions)

```
error: build of '/nix/store/nwdjlgsrncxl37k122w9qx3i14pn0rn9-env.drv' failed
https://github.com/NixOS/nixpkgs/pull/80458
1 package failed to build:
python38Packages.pytorch-metric-learning

1 package built:
python37Packages.pytorch-metric-learning
```

File: pkgs/development/python-modules/pytorch-metric-learning/default.nix
Comment (bcdarwin) [reply pkgs/development/python-modules/pytorch-metric-learning/default.nix]: Left without the extra negation for now.

**Comment by bcdarwin** [comment]
Everything resolved, package updated, now has tests.

**Comment by drewrisinger** [comment]
> Everything resolved, package updated, now has tests. python38Packages.pytorch is currently failing to build though.

Due to what? test failure? upstream dependency?

**Comment by drewrisinger** [comment]
@GrahamcOfBorg build python37Packages.pytorch-metric-learning
@GrahamcOfBorg build python38Packages.pytorch-metric-learning

File: pkgs/development/python-modules/pytorch-metric-learning/default.nix
Comment (bhipple) [inline pkgs/development/python-modules/pytorch-metric-learning/default.nix:35]:
```suggestion
    homepage = "https://github.com/KevinMusgrave/pytorch-metric-learning";
    changelog = "https://github.com/KevinMusgrave/pytorch-metric-learning/releases/tag/v${version}";
```
As a maintainer, one of the longer manual parts of reviewing update PRs is going and finding the releasenotes. Adding this attr makes that easier, since tools can look for it and link it automatically, e.g. https://github.com/ryantm/nixpkgs-update/pull/200

**Review by bhipple — CHANGES_REQUESTED** [review]
Minor nit, otherwise LGTM.

File: pkgs/development/python-modules/pytorch-metric-learning/default.nix
Comment (bcdarwin) [reply pkgs/development/python-modules/pytorch-metric-learning/default.nix:35]: Sure, added.

**Comment by bcdarwin** [comment]
@GrahamcOfBorg build python2Packages.pytorch-metric-learning python37Packages.pytorch-metric-learning python38Packages.pytorch-metric-learning

**Comment by bhipple** [comment]
Result of `nixpkgs-review pr 80458` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>20 packages marked as broken and skipped:</summary>
<br>- digitalbitbox
<br>- linuxPackages_4_4.evdi
<br>- linuxPackages_hardkernel_4_14.bcc
<br>- linuxPackages_hardkernel_4_14.bpftrace
<br>- linuxPackages_hardkernel_4_14.can-isotp
<br>- linuxPackages_hardkernel_4_14.chipsec
<br>- linuxPackages_hardkernel_4_14.digimend
<br>- linuxPackages_hardkernel_4_14.evdi
<br>- linuxPackages_hardkernel_4_14.mba6x_bl
<br>- linuxPackages_hardkernel_4_14.nvidia_x11
<br>- linuxPackages_hardkernel_4_14.nvidia_x11_legacy390
<br>- linuxPackages_hardkernel_4_14.nvidiabl
<br>- php73Extensions.zmq
<br>- php74Extensions.pcs
<br>- php74Extensions.pthreads
<br>- python27Packages.caffe
<br>- python27Packages.libmodulemd
<br>- python37Packages.pyblock
<br>- python38Packages.libselinux
<br>- python38Packages.pyblock
</details>
<details>
  <summary>2 packages built:</summary>
<br>- python37Packages.pytorch-metric-learning
<br>- python38Packages.pytorch-metric-learning
</details>

**Review by bhipple — APPROVED** [review]

**Pull request merged**
